.svglintrc.js

```diff
@@ -129,6 +131,34 @@ module.exports = {
               }
             }
           },
+          function(reporter, $, ast) {
+            reporter.name = "icon-precision";
+
+            const iconPath = $.find("path").attr("d");
+            const { segments } = parsePath(iconPath);
+            const segmentParts = segments.flat().filter((num) => (typeof num === 'number'));
```

Comment:
It seems that this custom check may fail here on NodeJS v10 or lower because `.flat()` is not a function. [The documentation for `.flat()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) suggests that we should be safe with NodeJS v11. So we need to at least update [`.travis.yml`](https://github.com/simple-icons/simple-icons/blob/develop/.travis.yml) to use `node_js: 11`. Can you update the `.travis.yml` file and push a change that should make Travis fail on this new check to verify that it does?

I'm also thinking of adding this detail to the Contributing Guidelines, thoughts on this (@mondeja as well)?

Reply:
All arrays returned by `parsePath` have 1 array in depth, so why not use this?

```suggestion
            const segmentParts = segments.reduce(
              (acc, val) => acc.concat(val), []
            ).filter((num) => (typeof num === 'number'));
```

Reply:
We can definitely do it manually, but I don't see a problem with updating the NodeJS version as it does not affect our NPM package.

I'm fine with upgrading to either 12 or 14 as well, definitely if 11 is not an LTS release.

Reply:
> I'm also thinking of adding this detail to the Contributing Guidelines...

What detail, the minor version of NodeJS required to run the linting process?

Reply:
> What detail, the minor version of NodeJS required to run the linting process?

Exactly, and maybe something similar for Jekyll and potentially Docker? Maybe other things as well? Something along the lines of a "Prerequisites" section... I'm unsure how to approach this as none of these are "hard" prerequisites.

Reply:
I don't think if this will be necessary if we use the stable versions of Jekyll and NodeJS, maybe only will add noise. But if we add it, simply pointing in "Building locally" section the list item "Make sure you have NodeJS v12 at least installed." at first should be enough.

- Building the Docker image locally you will see that `3.12` installs nodejs (12.18.4-r0) so is not necessary to add nothing to the Docker part.
- Jekyll, installed locally without specify a version will fetch the latest stable, which is what are we using for simple-icons.

Reply:
I do think it's useful to point out that NodeJS v12 is required, especially since svglint fails silently when a custom lint function errors (perhaps we should open an issue for that).

Regarding Docker, I don't imagine we use anything that is only supported in newer versions, but if so it would be useful to specify a minimum version as well. (for Docker I'm interested in whether you can build our docker image by running `docker build .`). As for Jekyll, I'm not too familiar with how people use it, but if someone still has an old version installed (for whatever reason), then they run into trouble. Although someone might figure this out from the output of Jekyll, having the Contributing guidelines just say that you need at least Jekyll v3.5 (following #3824) makes sense to me. For the same reason one would do it for NodeJS.

That said, this discussion (apart from including a minimum NodeJS version) should not block this PR from being merged!

Reply:
I've open #3959 to track this improvements separately. As for now, we can add at the start of the file this check to prevent silent failures:

```javascript
if (Array.prototype.flat === undefined) {
  console.error(`Minimum NodeJS v11.15.0 is required, but you are running ${process.version}.`);
  process.exit(1);
}
```

Reply:
My proposal for this would be to use the `engines` property of package.json and enforcing it using a `.npmrc`. This way, people wont be able to run `npm i` if the required version is not used.
I just commited the proposal.. Let me know what you think!

Reply:
> As for now, we can add at the start of the file this check to prevent silent failures [...]

Though not ideal, I'm not against it. In the meantime I opened https://github.com/birjolaxew/svglint/issues/20.

Reply:
> My proposal for this would be to use the engines property of package.json and enforcing it using a .npmrc. This way, people wont be able to run npm i if the required version is not used.
I just commited the proposal.. Let me know what you think!

I'm not 100% of this, but I think this also affects projects that install us as a dependency and the minimum NodeJS version for that is lower than v12 (not sure exactly what it is, I think the newest thing we use in the package is [`Object.defineProperty`](https://github.com/simple-icons/simple-icons/blob/2f5b6ee307479252c8b00b66999abc14c009ebf8/scripts/templates/index.js#L3)).

Reply:
> I'm not 100% of this, but I think this also affects projects that install us as a dependency and the minimum NodeJS version for that is lower than...

Yes, I've created a package as a test for `node: ">=16.0.0"` engine with `strict-engine=true` in `.npmrc` and this is the error trying to install it as dependency:

```
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: a-project-with-rare-name@1.0.0
npm ERR! notsup Not compatible with your version of node/npm: a-project-with-rare-name@1.0.0
npm ERR! notsup Required: {"node":">=16.0.0"}
npm ERR! notsup Actual:   {"npm":"7.0.8","node":"v15.2.1"}
```

> not sure exactly what it is...

- `Object.defineProperty` is [available since 0.12.18](https://node.green/#ES2015-built-ins-Symbol-Object-defineProperty-support).

Reply:
Oh, no problem! I will remove it and use @mondeja proposal for now.

Reply:
Alright :+1:  Now [Travis is failing because the NodeJS version is too low](https://travis-ci.com/github/simple-icons/simple-icons/jobs/443701618#L230), as expected. Can you update `.travis.yml` @service-paradis?


> Object.defineProperty is available since 0.12.18.

Hmm... :thinking: Then my memory is failing me :sweat_smile: I recall using a NodeJS feature that was fairly new at the time (like one or two LTSes before the latest at the time)... Anyway, this shouldn't block this PR. Would be nice if we _can_ specify a minimum NodeJS version in `package.json` at some point though.

Reply:
> Would be nice if we can specify a minimum NodeJS version in package.json at some point though.

I've included it in the TODO list of #3959 :+1:

Reply:
> Alright 👍 Now [Travis is failing because the NodeJS version is too low](https://travis-ci.com/github/simple-icons/simple-icons/jobs/443701618#L230), as expected. Can you update `.travis.yml`

Done :)

Reply:
Awesome, now [this build](https://travis-ci.com/github/simple-icons/simple-icons/jobs/443847121#L234) also shows that the check is able to fail on Travis. Thanks 👍

Reply:
Was pushing commits step by step to be sure :D